flink-core/src/main/java/org/apache/flink/api/common/functions/RichFunction.java

```diff
@@ -61,9 +61,30 @@ public interface RichFunction extends Function {
      *     When the runtime catches an exception, it aborts the task and lets the fail-over logic
      *     decide whether to retry the task execution.
      * @see org.apache.flink.configuration.Configuration
+     * @deprecated This method is deprecated since 1.18 and will be replaced by {@link
+     *     #open(OpenContext)}.
      */
+    @Deprecated
     void open(Configuration parameters) throws Exception;
 
+    /**
+     * Initialization method for the function. It is called before the actual working methods (like
+     * <i>map</i> or <i>join</i>) and thus suitable for one time setup work. For functions that are
+     * part of an iteration, this method will be invoked at the beginning of each iteration
+     * superstep.
+     *
+     * <p>By default, this method does nothing.
+     *
+     * @param openContext The context containing information about the context in which the function
+     *     is opened.
+     * @throws Exception Implementations may forward exceptions, which are caught by the runtime.
+     *     When the runtime catches an exception, it aborts the task and lets the fail-over logic
+     *     decide whether to retry the task execution.
+     */
+    default void open(OpenContext openContext) throws Exception {
+        open(new Configuration());
```

Comment:
I think we should explain that if not implemented, the default behavior will be calling the deprecated open method with an empty configuration.